react/README.md

```diff
@@ -18,3 +29,259 @@ For more info, see [Getting started with React](https://facebook.github.io/react
 [JSX](https://facebook.github.io/jsx/) is a JavaScript syntax extension that looks similar to XML. We use it with React because it is a concise and familiar syntax for defining tree structures with attributes.
 
 For more info, see [JSX in Depth](https://facebook.github.io/react/docs/jsx-in-depth.html).
+
+## General rules
+
+- Always use JSX syntax (don't use `React.createElement`)
+
+## Component files
+
+- Use PascalCase for React component names, e.g. `TextInput`
+- Use `.js` extension for React components
+- Use the component name for filenames. E.g., `TextInput.js`
+
+## Component class
+
+Prefer ES6 classes over `React.createClass` (eslint: [`react/prefer-es6-class`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-es6-class.md)):
+
+```jsx
+// good
+export default class MainComponent extends React.Component {
+
+}
+
+// bad (uses React.createClass)
+const MainComponent = React.createClass({
+
+});
+export default MainComponent
+```
+
+_NOTE:_ There is a common practice to use stateless/pure functions over ES6 classes for components without any internal state (eslint: [`react/prefer-stateless-function`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)), but we are choosing to stick with ES6 classes for a few reasons:
+
+- When the component does more than just render `props`, such as attaching callback handlers, the stateless function can become unnecessarily complex with nested functions
+- [`propTypes`](#component-propTypes) are defined _within_ the ES6 class, but have to be defined as additional properties _outside_ of a stateless function
+- Using ES6 classes for the main/default component help differentiate it from [helper components](#)
+
+## Component organization
+
+Export only one component per file as the default (eslint: [`react/no-multi-comp`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-multi-comp.md))
```

Comment:
I think there could be good reasons to export related components that belong to a suite.  For example, `<Modal>` and `<ModalFooter>`.  You could define the footer contents as a prop, but being able to compose the footer in JSX is more expressive.  Also, the footer has no context outside of `<Modal>`, so it makes sense to me to have them in the same module.  I also foresee nested `<Grid>` components (`<Grid>`, `<GridCell>`, `<GridSqueezePlop>`, etc) being very useful and expressive, but inherently coupled.